datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1390,8 +1399,9 @@ pub enum Partitioning {
     RoundRobinBatch(usize),
     /// Allocate rows based on a hash of one of more expressions and the specified number
     /// of partitions.
-    /// This partitioning scheme is not yet fully supported. See <https://issues.apache.org/jira/browse/ARROW-11011>
     Hash(Vec<Expr>, usize),
+    /// The DISTRIBUTE BY clause is used to repartition the data based on the input expressions
+    DistributeBy(Vec<Expr>),
```

Comment:
Another option here would be to re-use `Hash` and make the number of partitions an `Option<usize>`.

Reply:
Given the description in the spark docs https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-distribute-by.html I think either hashing or not would produce the same behavior with respect to partitioning.

However, using just the expressions as you have in this PR is probably more performant -- hashing the values would be wasted work.